docs/spec.md

```diff
@@ -0,0 +1,309 @@
+   </td>
+   <td>No
+   </td>
+   <td>Takes precedence over min and max values
+   </td>
+  </tr>
+</table>
+
+
+
+### Person Nodes
+
+
+<table>
+  <tr>
+   <td><strong>Field</strong>
+   </td>
+   <td><strong>Required</strong>
+   </td>
+   <td><strong>Notes</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>statementID
+   </td>
+   <td>Yes
+   </td>
+   <td>A unique ID for this node
+   </td>
+  </tr>
+  <tr>
+   <td>StatementType
+   </td>
+   <td>Yes
+   </td>
+   <td>Must be "personStatement" to generate person node
+   </td>
+  </tr>
+  <tr>
+   <td>names
+   </td>
+   <td>No
+   </td>
+   <td>Unknown person will be used if this data is missing. The name types will be used in the following order: 'individual', 'transliteration', 'alternative', 'birth', 'translation', 'former'. Unnamed person will be used if any specific name data is missing.
+   </td>
+  </tr>
+  <tr>
+   <td>names.fullName
+   </td>
+   <td>No
+   </td>
+   <td>This field takes precedence
+   </td>
+  </tr>
+  <tr>
+   <td>names.givenName
+   </td>
+   <td>No
+   </td>
+   <td>The name is a join of [givenName, patronymicName, familyName] if fullName is omitted
+   </td>
+  </tr>
+  <tr>
+   <td>names.patronymicName
+   </td>
+   <td>No
+   </td>
+   <td>The name is a join of [givenName, patronymicName, familyName] if fullName is omitted
+   </td>
+  </tr>
+  <tr>
+   <td>names.familyName
+   </td>
+   <td>No
+   </td>
+   <td>The name is a join of [givenName, patronymicName, familyName] if fullName is omitted
+   </td>
+  </tr>
+  <tr>
+   <td>personType
+   </td>
+   <td>No
+   </td>
+   <td>Unknown person will be assumed if the personType is missing.
+   </td>
+  </tr>
+  <tr>
+   <td>nationalities
+   </td>
+   <td>No
+   </td>
+   <td>No flags will be drawn if these data are missing. The visualiser only uses the first flag in the nationalities array.
+   </td>
+  </tr>
+  <tr>
+   <td>nationalities[0].code
+   </td>
+   <td>No
+   </td>
+   <td>No flag will be drawn if the country code is missing
+   </td>
+  </tr>
+</table>
+
+
+
+### Entity Nodes
+
+
+<table>
+  <tr>
+   <td><strong>Field</strong>
+   </td>
+   <td><strong>Required</strong>
+   </td>
+   <td><strong>Notes</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>statementID
+   </td>
+   <td>Yes
+   </td>
+   <td>Unique ID for this node
+   </td>
+  </tr>
+  <tr>
+   <td>statementType
+   </td>
+   <td>Yes
+   </td>
+   <td>Must be 'entityStatement' for entity node
+   </td>
+  </tr>
+  <tr>
+   <td>name
+   </td>
+   <td>No
+   </td>
+   <td>No label shown if missing
+   </td>
+  </tr>
+  <tr>
+   <td>entityType
+   </td>
+   <td>No
+   </td>
+   <td>Assumes unknown node type when entityType is missing
+   </td>
+  </tr>
+  <tr>
+   <td>publicListing
+   </td>
+   <td>No
+   </td>
+   <td>Assumes entity is not listed if this field is missing
+   </td>
+  </tr>
+  <tr>
+   <td>incorporatedInJurisdiction [jurisdiction]
+   </td>
+   <td>No
+   </td>
+   <td>No flags are drawn if this field is missing
```

Comment:
I think we need to add a row for `jurisdiction.code` (assuming this is what you use to generate the entity flags) below this as per the nationalities object?